### PR TITLE
fix: upgrade dependent image from docker.io to mcr

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

This pull request primarily focuses on updating the base image used in various Dockerfiles throughout the `sdk/ml/azure-ai-ml/tests/test_configs` directory. The base image has been changed from `ubuntu:16.04` to `mcr.microsoft.com/cbl-mariner/base/core:2.0`. This change affects the following Dockerfiles:

* [`sdk/ml/azure-ai-ml/tests/test_configs/components/additional_includes/docker/DockerFile`](diffhunk://#diff-11c77bb1891ab735fcd24c747beeb19c6280300e135b599b9d4784038a329648L1-R1): Updated the base image.
* [`sdk/ml/azure-ai-ml/tests/test_configs/deployments/batch/environment_files/DockerfileNonDefault`](diffhunk://#diff-11c77bb1891ab735fcd24c747beeb19c6280300e135b599b9d4784038a329648L1-R1): Updated the base image.
* [`sdk/ml/azure-ai-ml/tests/test_configs/environment/environment_files/DockerfileNonDefault`](diffhunk://#diff-11c77bb1891ab735fcd24c747beeb19c6280300e135b599b9d4784038a329648L1-R1): Updated the base image.
* [`sdk/ml/azure-ai-ml/tests/test_configs/internal/additional_includes/docker/DockerFile`](diffhunk://#diff-11c77bb1891ab735fcd24c747beeb19c6280300e135b599b9d4784038a329648L1-R1): Updated the base image.
* [`sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/my_exp/docker/DockerFile`](diffhunk://#diff-11c77bb1891ab735fcd24c747beeb19c6280300e135b599b9d4784038a329648L1-R1): Updated the base image.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/test_configs/components/additional_includes/docker/DockerFile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04
+FROM ubuntu:22.04
```
Comment (scbedd) [inline sdk/ml/azure-ai-ml/tests/test_configs/components/additional_includes/docker/DockerFile:1]: This needs to go to a non docker.io source.!

**Review by scbedd — CHANGES_REQUESTED** [review]
If we're going to touch these files, can we please resolve #34700?

These look like fairly simple images, it wouldn't be too hard to convert to mariner or an mcr ubuntu image right?

File: sdk/ml/azure-ai-ml/tests/test_configs/components/additional_includes/docker/DockerFile (hunk 1 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/test_configs/components/additional_includes/docker/DockerFile]: switch to mcr.microsoft.com/cbl-mariner/base/core:2.0

**Comment by elliotzh** [comment]
@scbedd switched to mcr, please help review

**Review by scbedd — APPROVED** [review]
❤️ ❤️ ❤️ 

🚀 

Ship it if it all works for you! Thank you so much @elliotzh !

**Pull request merged**